geniza/corpus/admin.py

```diff
@@ -143,6 +142,34 @@ def get_queryset(self, request):
             .annotate(shelfmk_all=ArrayAgg('textblock__fragment__shelfmark')) \
             .order_by('shelfmk_all')
 
+    def change_view(self, request, **kwargs):
+        """Add this document's first and last log entries to context.
+        
+        If there was a legacy date included in the log entry, show that info
+        instead of the timestamp of the entry itself.
+        """
+
+        # get first/last log entries; we want the reverse of their default
+        # ordering which is chronological descending
+        obj = Document.objects.get(pk=kwargs["object_id"])
+        first_input = obj.log_entries.last()
+        last_revision = obj.log_entries.first()
+
+        # check for "dated " message in the log entry and add legacy date.
+        # template will use the actual timestamp as a fallback if no legacy date
+        for log in first_input, last_revision:
+            try:
+                log.legacy_date = log.get_change_message().split("dated ")[1]
+            except IndexError:
+                continue
```

Comment:
Not sure it is worth this effort — can we just display the date in the date field _and_ revision message? It seems like knowing what was changed would be useful in every case.

Reply:
Sure, I like that idea! It only really makes sense for last revision though, right? The message would always be the same for first input, since there's only one kind of "first input" event.

Reply:
Oh, although I guess we might want to indicate ones that were first input into the spreadsheet vs. created in the database.